storage/src/model.rs

```diff
@@ -0,0 +1,3967 @@
+    pub project_number: Option<String>,
+    /// The bucket's retention policy. The retention policy enforces a
+    /// minimum retention time for all objects contained in the bucket, based
+    /// on their creation time. Any attempt to overwrite or delete objects
+    /// younger than the retention period will result in a PERMISSION_DENIED
+    /// error. An unlocked retention policy can be modified or removed from
+    /// the bucket via a storage.buckets.update operation. A locked retention
+    /// policy cannot be removed or shortened in duration for the lifetime of
+    /// the bucket. Attempting to remove or decrease period of a locked
+    /// retention policy will result in a PERMISSION_DENIED error.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub retention_policy: Option<BucketRetentionPolicy>,
+    /// The Recovery Point Objective (RPO) of this bucket. Set to ASYNC_TURBO
+    /// to turn on Turbo Replication on a bucket.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rpo: Option<String>,
+    /// Reserved for future use.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub satisfies_p_z_s: Option<bool>,
+    /// The URI of this bucket.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub self_link: Option<String>,
+    /// The bucket's default storage class, used whenever no storageClass is
+    /// specified for a newly-created object. This defines how objects in the
+    /// bucket are stored and determines the SLA and the cost of storage.
+    /// Values include MULTI_REGIONAL, REGIONAL, STANDARD, NEARLINE,
+    /// COLDLINE, ARCHIVE, and DURABLE_REDUCED_AVAILABILITY. If this value is
+    /// not specified when the bucket is created, it will default to
+    /// STANDARD. For more information, see storage classes.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub storage_class: Option<String>,
+    /// The creation time of the bucket in RFC 3339 format.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub time_created: Option<String>,
+    /// The modification time of the bucket in RFC 3339 format.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub updated: Option<String>,
+    /// The bucket's versioning configuration.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub versioning: Option<BucketVersioning>,
+    /// The bucket's website configuration, controlling how the service
+    /// behaves when accessing bucket contents as a web site. See the Static
+    /// Website Examples for more information.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub website: Option<BucketWebsite>,
+}
+
+impl Bucket {
+    /// Creates a builder to more easily construct the [Bucket] struct.
+    pub fn builder() -> BucketBuilder {
+        Default::default()
+    }
+}
+
+#[non_exhaustive]
+#[derive(Clone, Debug, Default)]
+/// A builder used to more easily construct the [Bucket] struct.
+pub struct BucketBuilder {
```

Comment:
These builders can never fail to build. Instead of builders should we just have setters on the original structs? Would save a lot of generated code. Thoughts?

Reply:
Sounds good but not urgent either way.  Maybe this is a question the community can help answer later.

Reply:
KK, will leave for now.